components/browser/toolbar/src/main/java/mozilla/components/browser/toolbar/BrowserToolbar.kt

```diff
@@ -330,6 +329,13 @@ class BrowserToolbar @JvmOverloads constructor(
         displayToolbar.addNavigationAction(action)
     }
 
+    /**
+     * Removes all added actions on the left, right sides and the menu in display mode.
+     */
+    override fun removeAllActions() {
+        displayToolbar.removeAllActions()
+    }
```

Comment:
This is going to be a problem!

You can't know what the app added and whether the app wants those actions be removed in a custom tab (That decision is probably up to the app). Imagine a tablet layout where you have back/forward buttons in the toolbar. Maybe we want to keep those. Chrome for example shows a bookmark icon in custom tabs on a tablet. Your code may remove that if it's not happening in or after `CustomTabsToolbarFeature`.

Regarding the "tabs tray" button. I think we should make `TabsToolbarFeature` smarter to not add a button if it's not needed - instead of adding it and then removing all again.

Reply:
Fair point!

The reason I also added this is when we support adding browser actions from a `CustomTabConfig` (e.g. a share icon instead of a menu option) I needed to do some clean up. This might need rethinking after consider that we might be adding ids to the menu items.